snapcraft/internal/lxd.py

```diff
@@ -152,10 +152,9 @@ def __init__(self, *, output, source, project_options,
                          remote=remote)
 
     def _ensure_container(self):
-        try:
-            check_call([
-                'lxc', 'start', self._container_name])
-        except CalledProcessError:
+        containers = check_output([
+            'lxc', 'list', self._container_name])
```

Comment:
smart, I did not think this would act as a filter, we just need to be careful in the `in` check below
```
$ lxc list snapcraft
+----------------+---------+----------------------+-----------------------------------------------+------------+-----------+
|      NAME      |  STATE  |         IPV4         |                     IPV6                      |    TYPE    | SNAPSHOTS |
+----------------+---------+----------------------+-----------------------------------------------+------------+-----------+
| snapcraft      | RUNNING | 10.120.128.94 (eth0) | fd42:40b3:c2e2:92a2:216:3eff:fe46:a89f (eth0) | PERSISTENT | 0         |
+----------------+---------+----------------------+-----------------------------------------------+------------+-----------+
| snapcraft-godd | STOPPED |                      |                                               | PERSISTENT | 0         |
+----------------+---------+----------------------+-----------------------------------------------+------------+-----------+
```